common/djangoapps/enrollment/views.py

```diff
@@ -515,9 +516,17 @@ def post(self, request):
         go through `add_enrollment()`, which allows creation of new and reactivation of old enrollments.
         """
         # Get the User, Course ID, and Mode from the request.
+        user = request.data.get('user', request.user.username)
+
+        email_pattern = r'^\w+[\.\w\+]*\w+@\w+\.\w{2,3}$'
```

Comment:
Please use email validator and try except block to support all emails supported  by open edx